### DISCUSSION TITLE
Only show the favorite charts in a dashboard on user level

### DISCUSSION THREAD

**Discussion by kevinwen2k** [apache/superset#18538] [Ideas]
**Is your feature request related to a problem? Please describe.**
We maintain a dashboard for a project. We have 20+ projects ongoing. We usually have 500+ charts on each of the projects. We have to use the tabs in each of dashboards to group charts together that lie on the similar interests. 

Usually, we have 40+ charts in one tab. However, the people who look at the same tab in the same project (dashboard) might not need to be interested in all of 40 charts. They might just be interested in some of them. So having the favorite charts in user level in a dashboard would be very helpful for us. It's even more beneficial to our customers when we turn on the alert and reporting so that our customers can receive the charts that they like in email.

**Describe the solution you'd like**
On each of the dashboards that users allow to view, If user add one of the charts in favorite, then the dashboard will show the favorite charts only. This favorite charts might be different from those favorite charts on chart list though.

**Describe alternatives you've considered**
The favorite charts on charts list could be combined with the favorite charts within a dashboard. It would be up to your team to decide the best solution

**Additional context**

**Comment by amitmiran137** [comment]
You could use DASHBOARD_RBAC feature flag and the assign users to roles and roles to dashboard for user to be able to access specific dashboards

Other then that, favourite are just an application attribute on top of access

**Comment by amitmiran137** [comment]
Now I see you were asking for charts .
Such functionality does not exist and you'd better separate to different dashboards logically

**Comment by amitmiran137** [comment]
@graceguo-supercat @etr2460 
You are familiar with such a setup mentioned above , maybe you can shed light on how this is done on your side ?

**Comment by graceguo-supercat** [comment]
Hi @kevinwen2k @amitmiran137, in airbnb we have some use cases like a large dashboard shared by different teams with 500+ unique users per week. We suggest dashboard builders to limit the number of charts per tab. For dashboard with 40+ charts in one dashboard view, we noticed the dashboard performance will be really bad. If user apply dashboard filters and filter is applicable to all charts in the current view, the perf is even worse and may freeze browser.

I don't think the solution that makes dashboard show favorite charts will work. This problem is more like how to organize large number of charts (or large amount of data), simply _hide and show inside a single dashboard_ can't handle it completely.

- Tab(s) and nested tabs are supposed to organize data inside dashboard. If each tab has  a lot charts and impacts performance, please consider create multiple dashboards.
- another use case is, all users shared a big set of charts, but each user only care a small set of charts, and each users' interested charts are not overlap. You should consider let user to build their own dashboard, but pick charts from the same pool. So that the same chart will be re-used in different dashboards. And once the chart is updated, all different dashboard should show this change accordingly.
